src/operator/numpy/np_broadcast_reduce_op.cuh

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2015-2020 by Contributors
+ * \file np_broadcast_reduce-inl.cuh
+ * \brief GPU implementations for numpy binary broadcast ops
+ * \author Zhaoqi Zhu
+*/
+#ifndef MXNET_OPERATOR_NUMPY_NP_BROADCAST_REDUCE_OP_CUH_
+#define MXNET_OPERATOR_NUMPY_NP_BROADCAST_REDUCE_OP_CUH_
+
+using namespace mshadow::cuda;
+using namespace mshadow;
+using namespace broadcast;
+
+#define KERNEL_UNROLL_SWITCH(do_unroll, unrollAmount, unrollVar, ...) \
+  if (do_unroll) {                                                    \
+    const int unrollVar = unrollAmount;                               \
+    {__VA_ARGS__}                                                     \
+  } else {                                                            \
+    const int unrollVar = 1;                                          \
+    {__VA_ARGS__}                                                     \
+  }
+
+template<typename Reducer, int NDim, typename DType, typename OType>
+void NumpyArgMinMaxReduce(Stream<gpu> *s, const TBlob& in_data, const TBlob& out_data,
```

Comment:
How is this function different from existing `Reduce`?

Reply:
Existing reduce uses MSHADOW_TYPE_SWITCH which only supports the mshadow data types. Since I am using a custom struct as accumulation type and output type I unrolled that switch into this custom reduce function

Reply:
Yup, that would be better - reuse is pretty much always better than duplication ;-).

Reply:
I am currently preparing a PR to refactor reductions to use RTC anyway, so having this logic in less places will also help me in that effort.

Reply:
Just applied that change